e2e/test_lint_requirements.sh

```diff
@@ -0,0 +1,18 @@
+#!/bin/bash
+# -*- indent-tabs-mode: nil; tab-width: 2; sh-indentation: 2; -*-
+
+# This script acts as an e2e test for lint_requirements command of fromager
+
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+source "$SCRIPTDIR/common.sh"
+
+pass=true
+
+fromager lint-requirements "$SCRIPTDIR/validate_inputs/constraints.txt" "$SCRIPTDIR/validate_inputs/requirements.txt"
+
+if [ $? -ne 0 ]; then
+    echo "the input files are not correctly formatted."
+    pass=false
+fi;
```

Comment:
Lets add some failing tests as well:

- At least one test for invalid constraints and/or requirements.  For this you can just add new files within `validate_inputs/` that mention in the filename and/or comments that they're purposefully bad.
- At least one test for invalid arguments to fromager.  Since it expects both constraints and requirements, maybe leave off one or another.

If you have other ideas for failing tests, feel free to add more.

For those tests you should be able to continue the same pattern of running fromager and checking `$?` and updating `pass` if any of the tests fail.  However, since we want the new tests to fail, change the `[ $? -ne 0 ]` to `[ $? -eq 0]`.

Reply:
Makes sense
I have added tests in latest push